File: AlgorithmProblems/1099. Two Sum Less Than K/main1099.cpp
class Solution1099 {
public:
    int twoSumLessThanK(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int i = 0, j = nums.size() - 1;
        int maxSum = -1;
        while (i < j) {
            int sum = nums[i] + nums[j];
            if (sum >= k) {
                j--;
            } else {
                maxSum = max(maxSum, sum);
                i++;
            }
        }
        return maxSum;
    }
};

File: AlgorithmProblems/1099. Two Sum Less Than K/main1099.py
class Solution1099:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()
        i, j = 0, len(nums) - 1
        res = -1
        while i < j:
            sum = nums[i] + nums[j]
            if sum >= k:
                j -= 1
            else:
                res = max(res, sum)
                i += 1
        return res